packages/cli/src/workflow-runner.ts

```diff
@@ -286,12 +295,54 @@ export class WorkflowRunner {
 					data.executionData,
 				);
 				workflowExecution = workflowExecute.processRunExecutionData(workflow);
+			} else if (data.preferredTrigger?.data && data.startNodes && !data.destinationNode) {
+				this.logger.debug(
+					`Execution ID ${executionId} had preferredTrigger. Starting from that trigger.`,
```

Comment:
(This is mostly me trying to make sense of this.) I see all the other branches specify _how much to execute_:

- has execution data (e.g. retry) → reuse that data
- has no run data or has no start node → run it fullly
- has run data or has start node → run it partially

But this nenw branch specifies _where to start executing from_. Should we stay consistent with the others? What would this branch be in the above terms? A "pseudo full" execution?

Reply:
This is what think it eventually should be:

```ts
let executionData
if (data.executionData) {
	executionData = data.executionData
} else if (data.preferredTrigger) {
	const triggerData = data.triggerData ?? getTriggerDataFromTrigger(data.preferredTrigger)
	executionData = createExecutionDataFromTrigger(data.preferredTrigger, data.triggerData, data.workflowData)
} else if (data.destinationNode) {
	executionData = recreateExecutionDataFromPartialExecution(data.destinationNode, data.workflowData, data.runData, data.pinData, ...etc)
}

// run with execution data
```

So you either start from old execution data, or from a trigger or we have the destination node and try to recreate the execution stack from that.

I don't even know if the block were I added the long TODO comment is used. I just don't want to remove it right now, but when we remove the old partial exec logic which comes with simplifying useRunWorkflow.ts to the point were I can be sure that the "Full Execution" branch is unused.

Reply:
Thank you, that's clearer.